Index: git/arch/arm/dts/imx8mm-evk.dts
===================================================================
--- git.orig/arch/arm/dts/imx8mm-evk.dts
+++ git/arch/arm/dts/imx8mm-evk.dts
@@ -277,6 +277,7 @@
 		};
 	};
 
+    /*
 	ptn5110: tcpc@50 {
 		compatible = "nxp,ptn5110";
 		pinctrl-names = "default";
@@ -318,6 +319,7 @@
 			self-powered;
 		};
 	};
+    */
 };
 
 &i2c3 {
@@ -358,12 +360,12 @@
 
 &usbotg1 {
 	status = "okay";
-	extcon = <&ptn5110>;
+	//extcon = <&ptn5110>;
 };
 
 &usbotg2 {
 	status = "okay";
-	extcon = <&ptn5110_2>;
+	//extcon = <&ptn5110_2>;
 };
 
 &usdhc2 {
Index: git/arch/arm/dts/imx8mm.dtsi
===================================================================
--- git.orig/arch/arm/dts/imx8mm.dtsi
+++ git/arch/arm/dts/imx8mm.dtsi
@@ -150,7 +150,7 @@
 
 	memory@40000000 {
 		device_type = "memory";
-		reg = <0x0 0x40000000 0 0x80000000>;
+		reg = <0x0 0x40000000 0 0xc0000000>;
 	};
 
 	osc_32k: clock-osc-32k {
Index: git/board/freescale/imx8mm_evk/lpddr4_timing.c
===================================================================
--- git.orig/board/freescale/imx8mm_evk/lpddr4_timing.c
+++ git/board/freescale/imx8mm_evk/lpddr4_timing.c
@@ -1,21 +1,22 @@
 /*
- * Copyright 2018-2019 NXP
+ * Copyright 2019 NXP
  *
  * SPDX-License-Identifier:	GPL-2.0+
  *
  * Generated code from MX8M_DDR_tool
+ * Align with uboot-imx_v2018.03_4.14.78_1.0.0_ga
  */
 
 #include <linux/kernel.h>
-#include <asm/arch/ddr.h>
+#include <asm/arch/imx8m_ddr.h>
 
 struct dram_cfg_param ddr_ddrc_cfg[] = {
-	/* Initialize DDRC registers */
+	/** Initialize DDRC registers **/
 	{ 0x3d400304, 0x1 },
 	{ 0x3d400030, 0x1 },
-	{ 0x3d400000, 0xa1080020 },
+	{ 0x3d400000, 0xa3080020 },
 	{ 0x3d400020, 0x223 },
-	{ 0x3d400024, 0x16e3600 },
+	{ 0x3d400024, 0x3a980 },
 	{ 0x3d400064, 0x5b00d2 },
 	{ 0x3d4000d0, 0xc00305ba },
 	{ 0x3d4000d4, 0x940000 },
@@ -45,17 +46,15 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d4001a8, 0x80000000 },
 	{ 0x3d4001b0, 0x11 },
 	{ 0x3d4001c0, 0x1 },
-	{ 0x3d4001c4, 0x0 },
+	{ 0x3d4001c4, 0x1 },
 	{ 0x3d4000f4, 0xc99 },
 	{ 0x3d400108, 0x70e1617 },
-	{ 0x3d400200, 0x1f },
+	{ 0x3d400200, 0x15 },
 	{ 0x3d40020c, 0x0 },
 	{ 0x3d400210, 0x1f1f },
 	{ 0x3d400204, 0x80808 },
 	{ 0x3d400214, 0x7070707 },
-	{ 0x3d400218, 0x7070707 },
-
-	/* performance setting */
+	{ 0x3d400218, 0x48080707 },
 	{ 0x3d400250, 0x29001701 },
 	{ 0x3d400254, 0x2c },
 	{ 0x3d40025c, 0x4000030 },
@@ -67,10 +66,8 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d400498, 0x620096 },
 	{ 0x3d40049c, 0x1100e07 },
 	{ 0x3d4004a0, 0xc8012c },
-
-	/* P1: 400mts */
 	{ 0x3d402020, 0x21 },
-	{ 0x3d402024, 0x30d400 },
+	{ 0x3d402024, 0x7d00 },
 	{ 0x3d402050, 0x20d040 },
 	{ 0x3d402064, 0xc001c },
 	{ 0x3d4020dc, 0x840000 },
@@ -93,10 +90,9 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d402190, 0x3818200 },
 	{ 0x3d402194, 0x80303 },
 	{ 0x3d4021b4, 0x100 },
-
-	/* p2: 100mts */
+	{ 0x3d4020f4, 0xc99 },
 	{ 0x3d403020, 0x21 },
-	{ 0x3d403024, 0xc3500 },
+	{ 0x3d403024, 0x1f40 },
 	{ 0x3d403050, 0x20d040 },
 	{ 0x3d403064, 0x30007 },
 	{ 0x3d4030dc, 0x840000 },
@@ -119,8 +115,7 @@ struct dram_cfg_param ddr_ddrc_cfg[] = {
 	{ 0x3d403190, 0x3818200 },
 	{ 0x3d403194, 0x80303 },
 	{ 0x3d4031b4, 0x100 },
-
-	/* default boot point */
+	{ 0x3d4030f4, 0xc99 },
 	{ 0x3d400028, 0x0 },
 };
 
@@ -208,8 +203,8 @@ struct dram_cfg_param ddr_ddrphy_cfg[] =
 	{ 0x220024, 0x1ab },
 	{ 0x2003a, 0x0 },
 	{ 0x20056, 0x3 },
-	{ 0x120056, 0xa },
-	{ 0x220056, 0xa },
+	{ 0x120056, 0x3 },
+	{ 0x220056, 0x3 },
 	{ 0x1004d, 0xe00 },
 	{ 0x1014d, 0xe00 },
 	{ 0x1104d, 0xe00 },
@@ -1060,8 +1055,7 @@ struct dram_cfg_param ddr_fsp0_cfg[] = {
 	{ 0x54008, 0x131f },
 	{ 0x54009, 0xc8 },
 	{ 0x5400b, 0x2 },
-	{ 0x5400d, 0x100 },
-	{ 0x54012, 0x110 },
+	{ 0x54012, 0x310 },
 	{ 0x54019, 0x2dd4 },
 	{ 0x5401a, 0x31 },
 	{ 0x5401b, 0x4d66 },
@@ -1073,7 +1067,7 @@ struct dram_cfg_param ddr_fsp0_cfg[] = {
 	{ 0x54022, 0x4d00 },
 	{ 0x54024, 0x16 },
 	{ 0x5402b, 0x1000 },
-	{ 0x5402c, 0x1 },
+	{ 0x5402c, 0x3 },
 	{ 0x54032, 0xd400 },
 	{ 0x54033, 0x312d },
 	{ 0x54034, 0x6600 },
@@ -1101,8 +1095,7 @@ struct dram_cfg_param ddr_fsp1_cfg[] = {
 	{ 0x54008, 0x121f },
 	{ 0x54009, 0xc8 },
 	{ 0x5400b, 0x2 },
-	{ 0x5400d, 0x100 },
-	{ 0x54012, 0x110 },
+	{ 0x54012, 0x310 },
 	{ 0x54019, 0x84 },
 	{ 0x5401a, 0x31 },
 	{ 0x5401b, 0x4d66 },
@@ -1114,7 +1107,7 @@ struct dram_cfg_param ddr_fsp1_cfg[] = {
 	{ 0x54022, 0x4d00 },
 	{ 0x54024, 0x16 },
 	{ 0x5402b, 0x1000 },
-	{ 0x5402c, 0x1 },
+	{ 0x5402c, 0x3 },
 	{ 0x54032, 0x8400 },
 	{ 0x54033, 0x3100 },
 	{ 0x54034, 0x6600 },
@@ -1142,8 +1135,7 @@ struct dram_cfg_param ddr_fsp2_cfg[] = {
 	{ 0x54008, 0x121f },
 	{ 0x54009, 0xc8 },
 	{ 0x5400b, 0x2 },
-	{ 0x5400d, 0x100 },
-	{ 0x54012, 0x110 },
+	{ 0x54012, 0x310 },
 	{ 0x54019, 0x84 },
 	{ 0x5401a, 0x31 },
 	{ 0x5401b, 0x4d66 },
@@ -1155,7 +1147,7 @@ struct dram_cfg_param ddr_fsp2_cfg[] = {
 	{ 0x54022, 0x4d00 },
 	{ 0x54024, 0x16 },
 	{ 0x5402b, 0x1000 },
-	{ 0x5402c, 0x1 },
+	{ 0x5402c, 0x3 },
 	{ 0x54032, 0x8400 },
 	{ 0x54033, 0x3100 },
 	{ 0x54034, 0x6600 },
@@ -1182,9 +1174,10 @@ struct dram_cfg_param ddr_fsp0_2d_cfg[]
 	{ 0x54008, 0x61 },
 	{ 0x54009, 0xc8 },
 	{ 0x5400b, 0x2 },
+	{ 0x5400d, 0x100 },
 	{ 0x5400f, 0x100 },
 	{ 0x54010, 0x1f7f },
-	{ 0x54012, 0x110 },
+	{ 0x54012, 0x310 },
 	{ 0x54019, 0x2dd4 },
 	{ 0x5401a, 0x31 },
 	{ 0x5401b, 0x4d66 },
@@ -1196,7 +1189,7 @@ struct dram_cfg_param ddr_fsp0_2d_cfg[]
 	{ 0x54022, 0x4d00 },
 	{ 0x54024, 0x16 },
 	{ 0x5402b, 0x1000 },
-	{ 0x5402c, 0x1 },
+	{ 0x5402c, 0x3 },
 	{ 0x54032, 0xd400 },
 	{ 0x54033, 0x312d },
 	{ 0x54034, 0x6600 },
@@ -1724,10 +1717,6 @@ struct dram_cfg_param ddr_phy_pie[] = {
 	{ 0x90013, 0x6152 },
 	{ 0x20010, 0x5a },
 	{ 0x20011, 0x3 },
-	{ 0x120010, 0x5a },
-	{ 0x120011, 0x3 },
-	{ 0x220010, 0x5a },
-	{ 0x220011, 0x3 },
 	{ 0x40080, 0xe0 },
 	{ 0x40081, 0x12 },
 	{ 0x40082, 0xe0 },
@@ -1853,3 +1842,4 @@ struct dram_timing_info dram_timing = {
 	.ddrphy_pie_num = ARRAY_SIZE(ddr_phy_pie),
 	.fsp_table = { 3000, 400, 100, },
 };
+
Index: git/configs/imx8mm_evk_defconfig
===================================================================
--- git.orig/configs/imx8mm_evk_defconfig
+++ git/configs/imx8mm_evk_defconfig
@@ -112,7 +112,6 @@ CONFIG_SYSRESET=y
 CONFIG_SYSRESET_PSCI=y
 CONFIG_DM_THERMAL=y
 CONFIG_NXP_TMU=y
-CONFIG_USB_TCPC=y
 CONFIG_USB=y
 CONFIG_USB_GADGET=y
 CONFIG_DM_USB=y
Index: git/include/configs/imx8mm_evk.h
===================================================================
--- git.orig/include/configs/imx8mm_evk.h
+++ git/include/configs/imx8mm_evk.h
@@ -207,7 +207,7 @@
 
 #define CONFIG_SYS_SDRAM_BASE           0x40000000
 #define PHYS_SDRAM                      0x40000000
-#define PHYS_SDRAM_SIZE			0x80000000 /* 2GB DDR */
+#define PHYS_SDRAM_SIZE			0xC0000000 /* 3GB DDR */
 
 #define CONFIG_SYS_MEMTEST_START	PHYS_SDRAM
 #define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + (PHYS_SDRAM_SIZE >> 1))
