diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts
index 618caf8baf6c..889274d0e804 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts
@@ -94,13 +94,23 @@
 	};
 };
 
-&usdhc3 {
-	status = "disabled";
-};
+/*&usdhc3 {
+//	status = "disabled";
+
+        pinctrl-names = "default", "state_100mhz", "state_200mhz";
+        pinctrl-0 = <&pinctrl_usdhc3>;
+        pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
+        pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
+        bus-width = <8>;
+        non-removable;
+        status = "okay";
 
-&flexspi {
+
+};*/
+
+/*&flexspi {
 	status = "disabled";
-};
+};*/
 
 /*
  * External OSC is used as PCIe REFCLK on RevC board.
@@ -108,6 +118,6 @@
  * PCIe REFCLK to internal PLL.
  */
 &pcie0{
-	ext_osc = <0>;
+	ext_osc = <0>;  // 0 - internal clock; 1 - external
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index 57de82ec4e76..71a9f9b50d19 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -89,6 +89,7 @@
 		};
 	};
 
+/*
 	wm8524: wm8524 {
 		compatible = "wlf,wm8524";
 		clocks = <&clk IMX8MM_CLK_SAI3_ROOT>;
@@ -96,7 +97,7 @@
 		wlf,mute-gpios = <&gpio5 21 GPIO_ACTIVE_LOW>;
 	};
 
-	sound-wm8524 {
+/*	sound-wm8524 {
 		compatible = "fsl,imx-audio-wm8524";
 		model = "wm8524-audio";
 		audio-cpu = <&sai3>;
@@ -104,17 +105,17 @@
 		audio-routing =
 			"Line Out Jack", "LINEVOUTL",
 			"Line Out Jack", "LINEVOUTR";
-	};
+	};*/
 
-	sound-ak4458 {
+/*	sound-ak4458 {
 		compatible = "fsl,imx-audio-ak4458";
 		model = "ak4458-audio";
 		audio-cpu = <&sai1>;
 		audio-codec = <&ak4458_1>, <&ak4458_2>;
 		ak4458,pdn-gpio = <&pca6416 4 GPIO_ACTIVE_HIGH>;
-	};
+	};*/
 
-	sound-ak5558 {
+/*	sound-ak5558 {
 		compatible = "fsl,imx-audio-ak5558";
 		model = "ak5558-audio";
 		audio-cpu = <&sai5>;
@@ -136,13 +137,15 @@
 		spdif-controller = <&spdif1>;
 		spdif-out;
 		spdif-in;
-	};
+	};*/
 
+	
+/*
 	sound-micfil {
 		compatible = "fsl,imx-audio-micfil";
 		model = "imx-audio-micfil";
 		cpu-dai = <&micfil>;
-	};
+	};*/
 };
 
 &clk {
@@ -263,13 +266,13 @@
 			>;
 		};
 
-		pinctrl_typec2: typec2grp {
+/*		pinctrl_typec2: typec2grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SD2_CD_B_GPIO2_IO12	0x159
 			>;
-		};
+		};*/
 
-		pinctrl_sai1: sai1grp {
+/*		pinctrl_sai1: sai1grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
 				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
@@ -284,9 +287,9 @@
 				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
 				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
 			>;
-		};
+		};*/
 
-		pinctrl_sai1_dsd: sai1grp_dsd {
+/*		pinctrl_sai1_dsd: sai1grp_dsd {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI1_MCLK_SAI1_MCLK	0xd6
 				MX8MM_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC	0xd6
@@ -301,9 +304,9 @@
 				MX8MM_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6	0xd6
 				MX8MM_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7	0xd6
 			>;
-		};
+		};*/
 
-		pinctrl_sai3: sai3grp {
+/*		pinctrl_sai3: sai3grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI3_TXFS_SAI3_TX_SYNC     0xd6
 				MX8MM_IOMUXC_SAI3_TXC_SAI3_TX_BCLK      0xd6
@@ -311,9 +314,9 @@
 				MX8MM_IOMUXC_SAI3_TXD_SAI3_TX_DATA0     0xd6
 				MX8MM_IOMUXC_I2C4_SDA_GPIO5_IO21        0xd6
 			>;
-		};
+		};*/
 
-		pinctrl_sai5: sai5grp {
+/*		pinctrl_sai5: sai5grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI5_MCLK_SAI5_MCLK	0xd6
 				MX8MM_IOMUXC_SAI5_RXC_SAI5_RX_BCLK	0xd6
@@ -323,7 +326,7 @@
 				MX8MM_IOMUXC_SAI5_RXD2_SAI5_RX_DATA2    0xd6
 				MX8MM_IOMUXC_SAI5_RXD3_SAI5_RX_DATA3    0xd6
 			>;
-		};
+		};*/
 
 		pinctrl_pdm: pdmgrp {
 			fsl,pins = <
@@ -695,25 +698,8 @@
 		};
 	};
 
-	typec1_ptn5110: tcpci@50 {
-		compatible = "usb,tcpci";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_typec1>;
-		reg = <0x50>;
-		interrupt-parent = <&gpio2>;
-		interrupts = <11 8>;
-		src-pdos = <0x380190c8>;
-		snk-pdos = <0x380190c8>;
-		/* Only can sink 5V for safe */
-		max-snk-mv = <5000>;
-		max-snk-ma = <3000>;
-		op-snk-mw = <10000>;
-		max-snk-mw = <15000>;
-		port-type = "drp";
-		default-role = "sink";
-		status = "okay";
-	};
-
+/*
+	// does not exist in this HW
 	typec2_ptn5110: tcpci@52 {
 		compatible = "usb,tcpci";
 		pinctrl-names = "default";
@@ -723,7 +709,7 @@
 		interrupts = <12 8>;
 		src-pdos = <0x380190c8>;
 		snk-pdos = <0x380190c8>;
-		/* Only can sink 5V for safe */
+		// Only can sink 5V for safe 
 		max-snk-mv = <5000>;
 		max-snk-ma = <3000>;
 		op-snk-mw = <10000>;
@@ -731,7 +717,7 @@
 		port-type = "drp";
 		default-role = "sink";
 		status = "okay";
-	};
+	};*/
 };
 
 
@@ -767,6 +753,29 @@
 		#gpio-cells = <2>;
 	};
 
+	// TCPC moved to I2C3 
+        typec1_ptn5110: tcpci@50 {
+                compatible = "usb,tcpci";
+                pinctrl-names = "default";
+                pinctrl-0 = <&pinctrl_typec1>;
+                reg = <0x50>;
+                interrupt-parent = <&gpio2>;
+                interrupts = <11 8>;
+                src-pdos = <0x380190c8>;
+                snk-pdos = <0x380190c8>;
+                /* Only can sink 5V for safe - fck it, sink up to 20V */
+                max-snk-mv = <20000>;
+                max-snk-ma = <3000>;
+                op-snk-mw = <10000>;
+                max-snk-mw = <15000>;
+                port-type = "drp";
+                default-role = "sink";
+                status = "okay";
+        };
+
+
+
+/*
 	ak4458_1: ak4458@10 {
 		compatible = "asahi-kasei,ak4458";
 		reg = <0x10>;
@@ -796,6 +805,7 @@
 		AVDD-supply = <&reg_audio_board>;
 		DVDD-supply = <&reg_audio_board>;
 	};
+*/
 
 	ov5640_mipi: ov5640_mipi@3c {
 		compatible = "ovti,ov5640_mipi";
@@ -839,6 +849,7 @@
 	status = "okay";
 };
 
+/*
 &sai1 {
 	pinctrl-names = "default", "dsd";
 	pinctrl-0 = <&pinctrl_sai1>;
@@ -884,6 +895,9 @@
 	status = "disabled";
 };
 
+*/
+
+/*
 &spdif1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_spdif1>;
@@ -901,6 +915,7 @@
 		"rxtx4", "rxtx5", "rxtx6", "rxtx7", "spba", "pll8k", "pll11k";
 	status = "okay";
 };
+*/
 
 &fec1 {
 	pinctrl-names = "default";
@@ -914,9 +929,9 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
-		ethphy0: ethernet-phy@0 {
+		ethphy0: ethernet-phy@4 {
 			compatible = "ethernet-phy-ieee802.3-c22";
-			reg = <0>;
+			reg = <4>;
 			at803x,led-act-blind-workaround;
 			at803x,eee-okay;
 			at803x,vddio-1p8v;
@@ -929,7 +944,9 @@
 	pinctrl-0 = <&pinctrl_pcie0>;
 	disable-gpio = <&gpio1 5 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio4 21 GPIO_ACTIVE_LOW>;
-	ext_osc = <1>;
+	//ext_osc = <1>;
+        ext_osc = <0>; // it seems we can use internal clock
+
 	status = "okay";
 };
 
@@ -967,11 +984,11 @@
 };
 
 &usbotg2 {
-	dr_mode = "otg";
-	extcon = <0>, <&typec2_ptn5110>;
-	picophy,pre-emp-curr-control = <3>;
-	picophy,dc-vol-level-adjust = <7>;
-	status = "disabled";
+	dr_mode = "host";
+	//extcon = <0>, <&typec2_ptn5110>;
+	//picophy,pre-emp-curr-control = <3>;
+	//picophy,dc-vol-level-adjust = <7>;
+	status = "okay";
 };
 
 &usdhc1 {
@@ -996,6 +1013,9 @@
 	bus-width = <4>;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
+        /delete-property/ mmc-pwrseq;
+        /delete-property/ cap-power-off-card;
+
 };
 
 &usdhc3 {
@@ -1035,6 +1055,8 @@
 	status = "okay";
 };
 
+
+/*
 &micfil {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pdm>;
@@ -1043,3 +1065,4 @@
 	assigned-clock-rates = <0>, <196608000>;
 	status = "okay";
 };
+*/
